app/screens/channel_info/channel_info.js

```diff
@@ -277,6 +283,37 @@ export default class ChannelInfo extends PureComponent {
                     this.close(false);
                 }
             };
+        } else if (eventType === 'undelete') {
```

Comment:
you are right the onPressAction had a lot of similar logic.  as I was refactoring. I realize the logic should not be the same. I removed else if and else statements in undelete because after it is archived. We are not navigating away. But in archive we are navigating away.